README.md

```diff
@@ -0,0 +1,186 @@
+
+```yaml
+kind: Service
+metadata:
+  name: SERVICE-NAME
+spec:
+  type: ClusterIP
+  clusterIP: None
+```
+
+#### Hazelcast Configuration
+
+The Hazelcast configuration to use DNS Lookup looks as follows.
+
+```xml
+<hazelcast>
+  <network>
+    <join>
+      <!-- deactivate normal discovery -->
+      <multicast enabled="false"/>
+      <tcp-ip enabled="false" />
+
+      <!-- activate the Kubernetes plugin -->
+      <discovery-strategies>
+        <discovery-strategy enabled="true"
+            class="com.hazelcast.kubernetes.HazelcastKubernetesDiscoveryStrategy">
+
+          <properties>
+            <property name="service-dns">MY-SERVICE-DNS-NAME</property>
+            <property name="service-dns-timeout">10</property>
+          </properties>
+        </discovery-strategy>
+      </discovery-strategies>
+    </join>
+  </network>
+</hazelcast>
+```
+
+There are 2 properties to configure the plugin:
+ * `service-dns` (required): service DNS, usually in the form of `SERVICE-NAME.NAMESPACE.svc.cluster.local`
+ * `service-dns-time` (optional): custom time for how long the DNS Lookup is checked
+
+**Note**: In this README, only XML configurations are presented, however you can achieve exactly the same effect using Java-based configurations.
+
+## Understanding Discovery Modes
+
+The following table summarizes the differences between the discovery modes: **Kubernetes API** and **DNS Lookup**.
+
+|                | Kubernetes API  | DNS Lookup |
+| -------------  | ------------- | ------------- |
+| Description    | Uses REST calls to Kubernetes Master to fetch IPs of PODs | Uses DNS to resolve IPs of PODs related to the given service |
+| Pros           | Flexible, supports **3 different options**: <br> - Hazelcast cluster per service<br> - Hazelcast cluster per multiple services (distinguished by labels)<br> - Hazelcast cluster per namespace | **No additional configuration** required, resolving DNS does not require granting any permissions  |
+| Cons           | Requires setting up **RoleBinding** (to allow access to Kubernetes API)  | - Limited to **headless Cluster IP** service<br> - Limited to **Hazelcast cluster per service**  |
+ 
+## Usage
+
+There are multiple scenarios of how the Hazelcast Kubernetes plugin can be used. The most popular use cases are presented below.
+
+### Embedded
+
+To use Hazelcast embedded in your application, you need to add the plugin dependency into your Maven/Gradle file. Then, when you provide `hazelcast.xml` as presented above or an equivalent Java-based configuration, your Hazelcast instances discover themselves automatically.
+
+#### Maven
+
+```xml
+<dependency>
+  <groupId>com.hazelcast</groupId>
+  <artifactId>hazelcast-kubernetes</artifactId>
+  <version>${hazelcast-kubernetes-version}</version>
+</dependency>
+```
+
+#### Gradle
+
+```groovy
+compile group: "com.hazelcast", name: "hazelcast-kubernetes", version: "${hazelcast-kubernetes-version}"
+```
```

Comment:
I suggest to move up maven/gradle section to Configuration...

Reply:
moved